tiledb/sm/array_schema/array_schema.h

```diff
@@ -77,7 +77,7 @@ class ArraySchema {
    *
    * @param array_schema The array schema to copy.
    */
-  explicit ArraySchema(const ArraySchema* array_schema);
+  explicit ArraySchema(const tdb_shared_ptr<const ArraySchema>& array_schema);
```

Comment:
From looking at this code previously, I know this constructor is only called once, in the C API function `tiledb_array_get_schema()` line 4772. As this PR changes allocation in that function, this constructor isn't called any more and can be removed.

Reply:
I still see this getting used btw.

Reply:
Which "this"? `tiledb_array_get_schema()` or `ArraySchema(...)`?

Reply:
Sorry, I am referring to the constructor. I attempted to remove it but it was still used in another place. I will punt for now.